### PR TITLE
Added a function to send telemetry to Quickwit's server.

### PR DESCRIPTION
**Pull request by fulmicoton**
This PR adds a telemetry sender.

By design it cannot send more than 10 events a minute and do one request per minute.
The information that is being send can be eyeball in a single file.
If it fails, it does so silently.

### Checklist
*Remove or complete this list if required.*
- [x] tested locally
Using npx http-echo-server and by hardcoding the URL to http://localhost:3000
- [x] added unit tests
- [x] included documentation

### CODE REVIEW

File: quickwit-telemetry/src/sender.rs [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+
+#[cfg(test)]
+impl ClockButton {
+    async fn tick(&self) {
+        let _ = self.0.send(()).await;
+    }
+}
+
+enum Clock {
+    Periocal(Mutex<Interval>),
```
Comment (evanxg852000) [inline quickwit-telemetry/src/sender.rs:54]: Maybe you meant `Periodical`

File: quickwit-telemetry/src/payload.rs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+use serde::{Deserialize, Serialize};
+use std::env;
+use std::time::UNIX_EPOCH;
+use uuid::Uuid;
+
+/// Represents the payload of the request send with telemetry requests.
```
Comment (evanxg852000) [inline quickwit-telemetry/src/payload.rs:26]: being nitpicky about `send` => `sent`?

File: quickwit-telemetry/src/sender.rs [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+            },
+        )
+    }
+
+    /// Adds an event.
+    /// If the queue is already saturated, (ie. it has reached the len `MAX_NUM_EVENTS_IN_QUEUE`)
+    // Returns true iff it was the first event in the queue.
+    fn push_event(&mut self, event: TelemetryEvent) -> bool {
+        if self.events.len() >= MAX_NUM_EVENTS_IN_QUEUE {
+            return false;
```
Comment (evanxg852000) [inline quickwit-telemetry/src/sender.rs:108]: probably forgot to  `self.num_dropped_events += 1`?

File: quickwit-telemetry/src/sender.rs (hunk 1 above)
Comment (fulmicoton) [reply quickwit-telemetry/src/sender.rs:40]: yes

File: quickwit-telemetry/src/sender.rs (hunk 3 above)
Comment (fulmicoton) [reply quickwit-telemetry/src/sender.rs:94]: oh yes nice catch.

**Pull request merged**